creator-node/src/premiumContent/helpers.ts

```diff
@@ -1,47 +1,51 @@
 import { signatureHasExpired } from '../apiSigning'
-import { PremiumContentSignatureData, PremiumContentType } from './types'
+import {
+  PremiumContentCIDResponse,
+  PremiumContentSignatureData,
+  PremiumContentType
+} from './types'
 import { getRegisteredDiscoveryNodes } from '../utils/getRegisteredDiscoveryNodes'
 import { Redis } from 'ioredis'
+import redis from '../redis'
 import type Logger from 'bunyan'
 
-const models = require('../models')
-const { QueryTypes } = require('sequelize')
-
 const PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+export const PREMIUM_CONTENT_CID_CACHE_KEY = 'premium-content-cids'
 
-/**
- * Return whether the CID is for a premium track by
- * checking if it's for a 'track' or 'copy320' file
- * and getting the track metadataJSON for the
- * track blockchain id to see if it is premium.
- */
-export async function isCIDForPremiumTrack(cid: string): Promise<
-  | {
-      trackId: null
-      isPremium: false
-    }
-  | {
-      trackId: number
-      isPremium: boolean
-    }
-> {
-  const result = await models.sequelize.query(
-    `select t.* from "Tracks" t, "Files" f
-      where f."multihash" = :cid
-      and f."type" in ('track', 'copy320')
-      and f."trackBlockchainId" = t."blockchainId"`,
-    {
-      replacements: { cid },
-      type: QueryTypes.SELECT
-    }
-  )
-  if (!result.length) {
+export async function isCIDForPremiumTrack({
+  cid,
+  redis
+}: {
+  cid: string
+  redis: Redis
+}): Promise<PremiumContentCIDResponse> {
+  const premiumContentCIDCache = await redis.get(PREMIUM_CONTENT_CID_CACHE_KEY)
```

Comment:
Redis has native set functionality that we should use for this rather than serializing a JSON map

Reply:
using `del` and `hset` now to set the cid -> trackId map, and `hget` with the cid to immediately retrieve the trackId value if the cid key exists in the map.
my 1 worry now is the edge case where there is a small amount of time between the cache being removed and being reset and users trying to access premium track during that split millisecond.

Reply:
if I'm understanding this edge case correctly it can be solved by removing the `del` since `hset` will override anyway - [these docs](https://redis.io/commands/hset/) say "If key does not exist, a new key holding a hash is created. If field already exists in the hash, it is overwritten"

Reply:
the issue is for keys that existed and that will not be overwritten... what if a track become non-premium, its CIDs need to be removed from the cache. if we dont `del`, those cids will remain in the cache and be treated as premium. in this case we'd have to do some extra linear computation on the whole cache to see which keys to remove.
does this make sense to you?

Reply:
thanks for clarifying - I get what you're saying if there are multiple keys to update, but I'm only seeing one key that gets deleted and then immediately set to a new mapping on lines 37+38 of `src/premiumContent/helpers.ts`. is there somewhere else where multiple keys need to be removed from the cache?

Reply:
so there is the redis cache key, and there is the cache value which is a dictionary. that dictionary is a key/value pair as you already know. sometimes we need to remove some keys from the dictionary, namely the cids that used to be premium but are no longer premium. only ways i can think of doing this are:
- use `del` as we are doing right now, but i mentioned the potential drawback in my 1st comment on this thread
- compare what the query returned and what's in the cache and remove the difference, which is what i mean by linear comparison

if stll unclear (or if im still missing your point), very happy to chat sync on this :)

Reply:
synced offline - I understand now because it's hset not set, and the new mapping can be missing some old keys that won't get removed otherwise 👍